python/ray/data/_internal/stats.py

```diff
@@ -263,7 +267,9 @@ def summary_string(
                     out += "\n"
         if len(self.stages) == 1:
             stage_name, metadata = next(iter(self.stages.items()))
-            stage_uuid = self.dataset_uuid + stage_name
+            # TODO(ekl) deprecate and remove the notion of dataset UUID once we move
+            # fully to streaming execution.
+            stage_uuid = (self.dataset_uuid or "unknown_uuid") + stage_name
```

Comment:
We have a hack in the followup PR to set this correctly. For now, adding a default so that we don't need to be adding dataset_uuid to the execute() API if we are removing it in the future.